odoo/models.py

```diff
@@ -5210,10 +5210,10 @@ def browse(self, ids=None):
         :type ids: int or iterable(int) or None
         :return: recordset
         """
-        if not ids:
-            ids = ()
-        elif ids.__class__ in IdType:
+        if ids.__class__ in IdType:
             ids = (ids,)
+        elif not ids:
+            ids = ()
```

Comment:
Agree for the benchmark.
But in any case this method is confusing for now : the `ids.__class__ in IdType` condition is not completely correct because the `NewId` is in `IdType` but it is also always Falsy, then a part of the `ids.__class__ in IdType` condition is not used at least.

Reply:
I made some benchmark, and you are perfectly right, it is **worst** with this commit:
**70 %** of browse call is with Falsy values (or None), **28 %** with Iterable values, and only **2 %** with Int or NewId values.
And because my change decrease the performance of the first case, it is not worth it (even if it increases the performance of others cases, in the total, it is slower).

I propose to remove this commit, and replace it with another changing the second condition 
`if ids.__class__ in IdType:` → `if ids.__class__ is int:` (or with isinstance). It is ok for you ?